index.js

```diff
@@ -190,7 +191,19 @@ function globIsSingular(glob) {
 }
 
 function getBasePath(ourGlob, opt) {
-  return resolveGlob(globParent(ourGlob) + path.sep, opt);
+  var basePath;
+  var parent = globParent(ourGlob);
+
+  if (parent === '/' && opt && opt.root) {
+    basePath = path.join(opt.root);
+  } else {
+    basePath = resolveGlob(parent, opt);
+  }
+
+  if (!sepRe.test(basePath.charAt(basePath.length - 1))) {
```

Comment:
> Is testing the last character better than a $ in the RegExp?

I think so because I don't think regex will start from the end of a string and test backwards so it'll be testing all the possible combinations leading up to the end. By only testing the last character, it reduces the amount of characters that the regex has to go through.